File: frontend/lib/models/api/inventory.dart
class Inventory {
  String poPireference;
  String invoiceNum;
  String itemCode;
  String itemDescription;
  int brandId;
  int packSize;
  int lotNum;
  String expiry;
  double costOfSale;
  String? note;
  int addedBy;
  String dateTimeAdded;
  int quantity;

  Inventory({
    required this.poPireference,
    required this.invoiceNum,
    required this.itemCode,
    required this.itemDescription,
    required this.brandId,
    required this.packSize,
    required this.lotNum,
    required this.expiry,
    required this.costOfSale,
    this.note,
    required this.addedBy,
    required this.dateTimeAdded,
    required this.quantity
  });

  factory Inventory.fromJson(Map<String, dynamic> json) {
    return Inventory(
      poPireference: json["poPireference"],
      invoiceNum: json["invoiceNum"],
      itemCode: json["itemCode"],
      itemDescription: json["itemDescription"],
      brandId: json["brandId"],
      packSize: json["packSize"],
      lotNum: json["lotNum"],
      expiry: json["expiry"],
      costOfSale: (json["costOfSale"] is int)
        ? (json["costOfSale"] as int).toDouble()
        : json["costOfSale"],
      note: json["note"],
      addedBy: json["addedBy"],
      dateTimeAdded: json["dateTimeAdded"],
      quantity: json["quantity"]
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "poPireference": poPireference,
      "invoiceNum": invoiceNum,
      "itemCode": itemCode,
      "itemDescription": itemDescription,
      "brandId": brandId,
      "packSize": packSize,
      "lotNum": lotNum,
      "expiry": expiry,
      "quantity": quantity,
      "costOfSale": costOfSale,
      "note": note,
      "addedBy": addedBy,
      "dateTimeAdded": dateTimeAdded,
    };
  }
}

File: lis/src/main/java/com/biotech/lis/Entity/Inventory.java
package com.biotech.lis.Entity;

import java.time.LocalDate;
import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "`inventory`")
public class Inventory {
    @Id
    private String itemCode;

    private String poPireference;
    private String invoiceNum;
    private String itemDescription;
    private Integer brandId;
    private Integer lotNum;
    private LocalDate expiry;
    private Integer packSize;
    private Integer quantity;
    private Double costOfSale;
    private String note;

    //note: do not display on frontend inventory tabl
    private Long addedBy;
    private LocalDateTime dateTimeAdded;
}
